<?xml version="1.0"?>
<launch>
  <!-- Override SLOAM parameters -->
  <param name="sloam/seg_model_path" value="/opt/sloam_ws/src/models/darknet53_segmentator.onnx" />
  <param name="sloam/cloud_topic" value="/forest_robot/velodyne_points" />
  <param name="sloam/odom_topic" value="/odom" />
  <param name="sloam/debug_mode" value="true" />
  <param name="sloam/do_destagger" value="false" />
  <param name="sloam/map_frame_id" value="map" />
  <param name="sloam/odom_frame_id" value="odom" />
  <param name="sloam/robot_frame_id" value="base_link" />
  
  <!-- Launch SLOAM segmentation node only -->
  <node pkg="sloam" type="sloam_inference_node" name="segmentation" output="screen" />
  
  <!-- Launch RViz for visualization -->
  <node pkg="rviz" type="rviz" name="sloam_rviz" output="screen" args="-d $(find sloam)/launch/rviz/segmentation.rviz"/>
  
  <!-- Add TF frame transforms -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_map" 
        args="0 0 0 0 0 0 world map 100" />
</launch>